jetty-client/src/main/java/org/eclipse/jetty/client/HttpReceiver.java

```diff
@@ -653,15 +653,22 @@ public String toString()
     {
         private final Map<Object, Long> demands = new ConcurrentHashMap<>();
         private final LongConsumer demand = HttpReceiver.this::demand;
-        private final List<Response.DemandedContentListener> listeners;
+        private final List<Response.DemandedContentListener> listeners = new ArrayList<>(2);
 
-        private ContentListeners(List<Response.ResponseListener> responseListeners)
+        private void clear()
         {
-            listeners = new ArrayList<>(responseListeners.size());
-            responseListeners.stream()
-                .filter(Response.DemandedContentListener.class::isInstance)
-                .map(Response.DemandedContentListener.class::cast)
-                .forEach(listeners::add);
+            demands.clear();
+            listeners.clear();
+        }
+
+        private void reset(List<Response.ResponseListener> responseListeners)
```

Comment:
This is more an `initialise` than a `reset`

Reply:
@gregw I see `initialize()` as a once-off method. Here we are re-setting every time this class is reused, so I like `reset()` better.

Reply:
how about `reinitialize` ?
Having a `clear` and a `reset` is just weird... so one of them must be wrong!

Reply:
I don't see why `clear` and `reset` are weird? 
One is to `clear` to avoid holding onto things, `reset` is "be ready again".
`reset` has been used this way in e.g. `IteratingCallback.reset()` and `HttpConnection.SendCallback.reset(...)`, here is the same.